### ISSUE TITLE
Expose chunk name to assets including siblings in toJson

### ISSUE THREAD

**Issue by slavab89** [webpack/webpack#7073]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
By using the new webpack 4 `splitChunks` functionality, we get chunks such as `vendors~main` or `vendors~chunkA~chunkB~chunkC` etc.
When trying to use SSR and return some dynamic chunk - chunkA - we need to return both chunkA asset itself and the `vendors~chunkA~chunkB~chunkC` asset as well.
With the current toJson() of stats its pretty hard to find all the related files and get them.

**If this is a feature request, what is motivation or use case for changing the behavior?**
Using `namedChunkGroups` under compilation result it is much easier to get all the relevant chunks and assets of the wanted one. For example:
```
const chunkNameToAssets = {};
namedChunkGroups.forEach((chunkGroup, name) => {
  const chunkFiles = _.flatten(chunkGroup.chunks.map(chunk => chunk.files));
  if (!_.isEmpty(chunkFiles)) {
    chunkNameToAssets[name] = chunkFiles;
  }
});
```
Would it be possible to expose that structure in some manner to `toJson` output?

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**
Node 8
webpack 4.5

**Comment by sokra** [comment]
Yep, send a PR adding this. You can take a look at the `entrypoints` property, which does the same but only for entrypoints.
